lib/lndclient/LndClient.ts

```diff
@@ -295,84 +295,64 @@ class LndClient extends SwapClient {
     return this.unaryCall<lndrpc.GetInfoRequest, lndrpc.GetInfoResponse>('getInfo', new lndrpc.GetInfoRequest());
   }
 
-  public sendSmallestAmount = async (rHash: string, destination: string) => {
-    const sendRequest = new lndrpc.SendRequest();
-    sendRequest.setAmt(1);
-    sendRequest.setDestString(destination);
-    sendRequest.setPaymentHashString(rHash);
-    sendRequest.setFinalCltvDelta(this.cltvDelta);
-    let sendPaymentResponse: lndrpc.SendResponse;
-    try {
-      sendPaymentResponse = await this.sendPaymentSync(sendRequest);
-    } catch (err) {
-      this.logger.error('got exception from sendPaymentSync', err.message);
-      throw err;
-    }
-    const paymentError = sendPaymentResponse.getPaymentError();
-    if (paymentError) {
-      throw new Error(paymentError);
-    }
-    return base64ToHex(sendPaymentResponse.getPaymentPreimage_asB64());
+  public sendSmallestAmount = async (rHash: string, destination: string): Promise<string> => {
+    const request = this.buildSendRequest({
+      rHash,
+      destination,
+      amount: 1,
+      // In case of sanity swaps we don't know the
+      // takerCltvDelta or the makerCltvDelta. Using our
+      // client's default.
+      finalCltvDelta: this.cltvDelta,
+    });
+    const preImage = await this.executeSendRequest(request);
+    return preImage;
   }
 
   public sendPayment = async (deal: SwapDeal): Promise<string> => {
     assert(deal.state === SwapState.Active);
-
-    if (deal.makerToTakerRoutes && deal.role === SwapRole.Maker) {
```

Comment:
Should we drop the `makerToTakerRoutes` field from the swap deal type?

Reply:
I'm not sure about this since it's still being used in `Swaps` to derive `makerCltvDelta`. If we decide to do this I'd do it in a separate PR.

Reply:
OK that's fine, it can be done separately.